pathfinderlib/main.py

```diff
@@ -25,33 +63,212 @@ def make_fastq_to_fasta_command(self, filename):
+    def __init__(self, mapping_method, search_method, evalue_cutoff, humann_fp):
+        super(Humann, self).__init__(mapping_method, search_method, evalue_cutoff)
+        self.humann_fp = humann_fp
+
+    def run(self, alignment_fp, output_dir):
+        # check the search method and make changes to the alignment file to work with humann
+
+        # softlink alignment files to it's input directory
+        #link_fp = os.path.join(self.humann_fp, "input", os.path.basename(alignment_fp)+'.txt')
+        #print(link_fp)
+        #print(alignment_fp)
+        #os.symlink(alignment_fp, link_fp)
+
+        # run the program
+        #os.chdir(self.humann_fp)
+        #subprocess.check_call("scons", stderr=subprocess.STDOUT)
         
-    def run(self, R1, R2, output):
-        self.run_blastx(R1,"_R1.txt", output)
-        self.run_blastx(R2, "_R2.txt", output)
-        os.chdir(self.config["humann_fp"])
-        subprocess.check_call("scons", stderr=subprocess.STDOUT)
+        # unlink the input files from it's input directory
+        #os.unlink(link_fp)
+
+        # move the output files to the designated output dir
+
+        # parse summary results
+        #summary = parseSummary()
+        #return summary
+        raise NotImplementedError("Humann is not yet implemented to work with the pipeline.")
+    
+    def _fix_alignment_result(self, alignment):
+        "Manipulates the alignment file to make it readable by Humann"
+        pass ##RAPsearch has 4 headerlines that humann doesn't seem to like...
+
+    def _parseSummary(alignment):
+        pass ## to implement later if we would like to pass the results to the summary file
+
+    def make_index(self):
+        pass
+
+    def index_exists(self):
+        return True
+        
+class BestHit(_Assignment):
+    def __init__(self, mapping_method, search_method, evalue_cutoff, kegg_fp, kegg_to_ko_fp):
+        super(BestHit, self).__init__(mapping_method, search_method, evalue_cutoff)
+        self.kegg_fp = kegg_fp
+        self.kegg_to_ko_fp = kegg_to_ko_fp
+        
+    def _getCount(self, df, group_key, count_key):
+        "Counts how many times each value is repeated in a group_key column."
+        return df.groupby(group_key).size().to_frame(name=count_key).reset_index()
+
+    def _sumAbundance(self, df, group_key, sum_by_key, count_key):
+        return df.groupby(group_key).apply(lambda g: g[sum_by_key].sum()).to_frame(name=count_key).reset_index()
+
+    def _load_kegg2ko(self):
+        "Loads the index file and counts how many KOs are assigned to each protein."
+        kegg2ko = pandas.read_csv(self.kegg_to_ko_fp, sep='\t', header=0) # Headers are Subject, KO
+        return kegg2ko.merge(self._getCount(kegg2ko, 'Subject', 'Count_ko'), on='Subject')
+
+    def _parseResults(self, alignment_fp):
+        "Parses an alignment result file. Returns only the columns of interest"
+        colNames = ['# Fields: Query', 'Subject', 'identity', 'e-value']
```

Comment:
OK, I see what's happening now.  We may refactor this in the future but let's leave it for now.